src/Compilers/VisualBasic/Portable/Compilation/VisualBasicCompilation.vb

```diff
@@ -1851,6 +1851,27 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
             Return diagnostic Is Nothing OrElse diagnostic.Severity <> DiagnosticSeverity.Error
         End Function
 
+        Private Protected Overrides Function IsSymbolAccessibleWithinCore(symbol As ISymbol, within As ISymbol, throughType As ITypeSymbol) As Boolean
+            Dim symbol0 = TryCast(symbol, Symbol)
+            If (symbol0 Is Nothing) Then
+                Throw New ArgumentException(String.Format(CodeAnalysisResources.IsSymbolAccessibleWrongAssembly, NameOf(symbol)), NameOf(symbol))
```

Comment:
> Throw New ArgumentException(String.Format(CodeAnalysisResources.IsSymbolAccessibleWrongAssembly, NameOf(symbol)), NameOf(symbol)) [](start = 15, length = 130)

Similar to C#, I think we should use EnsureVbSymbolOrNothing helper for checks like that.